keras_nlp/utils/text_generation.py

```diff
@@ -681,7 +692,18 @@ def token_probability_fn(inputs):
         next_token = tf.cast(next_token, dtype=prompt.dtype)
         # Append the next token to current sequence.
         prompt = tf.concat([prompt, next_token[:, tf.newaxis]], axis=-1)
-        i += 1
+        return [prompt]
+
+    # Run a while loop till text of length `max_length` has been generated.
+    prompt = tf.while_loop(
+        cond=lambda prompt: tf.less(
+            tf.cast(tf.shape(prompt)[-1], dtype=tf.int64),
+            tf.cast(max_length, dtype=tf.int64),
+        ),
+        body=one_step,
+        loop_vars=[prompt],
+        shape_invariants=[tf.TensorShape(shape_invariants)],
```

Comment:
Can we just pass `tf.TensorShape([None, None])` as the shape invariant? Generally we should support a static batch size of `None`, tf data does this by default after calling `.batch()` for example. Might simplify the code a bit.